.github/workflows/screenshots.yml

```diff
@@ -0,0 +1,178 @@
+name: Screenshots
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, labeled, unlabeled]
+
+env:
+  S3_BUCKET: ${{ secrets.S3_BUCKET }}
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  CONFIGURE_ENCRYPTION_KEY: ${{ secrets.CONFIGURE_ENCRYPTION_KEY }}
+
+jobs:
+  build:
+    name: Build Application
+    if: contains(github.event.pull_request.labels.*.name, 'Needs Screenshots')
+    runs-on: macos-latest
+
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-ruby@v1
+      with:
+        ruby-version: '2.6.x'
+    
+    - name: Install Bundler
+      run: gem install bundler
+    
+    - name: Restore Ruby Dependency Cache
+      id: restore-ruby-dependency-cache
+      uses: actions/cache@v1
+      with:
+        path: vendor/bundle
+        key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
+    
+    - name: Install Ruby Dependencies
+      run: |
+        bundle config path vendor/bundle
+        bundle install --jobs 4 --retry 3
```

Comment:
I was gonna suggest to use an `if` to check for `cache-hit` to skip running `bundle install` if the cache in the step above was hit (docs [here](https://github.com/actions/cache/tree/e8230b28a9d4dd887adc2d080b81482b2d47b185#skipping-steps-based-on-cache-hit), but I can see that both the [Ruby example](https://github.com/actions/cache/blob/e8230b28a9d4dd887adc2d080b81482b2d47b185/examples.md#ruby---bundler) and the [Actions docs](https://help.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows#example-using-the-cache-action) don't do it. Do you know why?

Reply:
I'm not 100% sure why, but I found that just restoring from cache would sometimes have weird failures (particularly on our other dependencies restoration). 

Allowing `bundle install` to run after restoring the cache only takes ~1 second, so whatever bookkeeping it's doing, it makes everything work :)

Reply:
Great to know! I'll update my setup as well then 👌